FirebaseAuth/Sources/Backend/FIRIdentityToolkitRequest.m

```diff
@@ -80,23 +84,37 @@ - (BOOL)containsPostBody {
 
 - (NSURL *)requestURL {
   NSString *apiURLFormat;
+  NSString *apiPrefix;
   NSString *apiHost;
+
+  NSString *emulatorURL = _requestConfiguration.emulatorURL;
+
   if (_useIdentityPlatform) {
     apiURLFormat = kIdentityPlatformAPIURLFormat;
-    if (_useStaging) {
+    apiPrefix = kHttpsPrefix;
+    if (emulatorURL) {
+      apiPrefix = kHttpPrefix;
+      apiHost =
+          [NSString stringWithFormat:kEmulatorHostFormat, emulatorURL, kIdentityPlatformAPIHost];
```

Comment:
I'd also suggest that we don't forcibly fit `localhost:9000/www.googleapis.com` into any variable named `host`. It's confusing since it's not a host (it is host + port + path prefix), and it is fragile since things may break if we ever refactor to use `CFURL` for URL creation. It's up to you though.

Reply:
I changed the variable to be apiHostAndPathPrefix for more clarity.